packages/google_maps_flutter/google_maps_flutter_ios/example/ios11/ios/RunnerTests/GoogleMapsPolylinesControllerTests.m

```diff
@@ -0,0 +1,56 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+@import google_maps_flutter_ios;
+@import google_maps_flutter_ios.Test;
+@import XCTest;
+@import GoogleMaps;
+
+#import <OCMock/OCMock.h>
+#import "PartiallyMockedMapView.h"
+
+@interface GoogleMapsPolylinesControllerTests : XCTestCase
+@property(strong, atomic) FLTGoogleMapPolylineController *polylineController;
+@end
+
+@implementation GoogleMapsPolylinesControllerTests
+
+- (void)setUp {
+  NSDictionary *polyline = @{
+    @"points" : @[
+      @[ @(52.4816), @(-3.1791) ], @[ @(54.043), @(-2.9925) ], @[ @(54.1396), @(-4.2739) ],
+      @[ @(53.4153), @(-4.0829) ]
+    ],
+    @"polylineId" : @"polyline_id_0",
+
+  };
+
+  GMSMutablePath *path = [FLTPolylinesController getPath:polyline];
+  NSString *identifier = polyline[@"polylineId"];
+
+  CGRect frame = CGRectMake(0, 0, 100, 100);
+  PartiallyMockedMapView *mapView = [[PartiallyMockedMapView alloc]
+      initWithFrame:frame
+             camera:[[GMSCameraPosition alloc] initWithLatitude:0 longitude:0 zoom:0]];
+
+  self.polylineController = [[FLTGoogleMapPolylineController alloc] initPolylineWithPath:path
+                                                                              identifier:identifier
+                                                                                 mapView:mapView];
+}
+
+- (void)testSetPatterns {
+  NSArray<GMSStrokeStyle *> *styles = @[
+    [GMSStrokeStyle solidColor:[UIColor clearColor]], [GMSStrokeStyle solidColor:[UIColor redColor]]
+  ];
+
+  NSArray<NSNumber *> *lengths = @[ @10, @10 ];
+
+  XCTAssertNil(self.polylineController.polyline.spans);
+
+  [self.polylineController setPattern:styles lengths:lengths];
+
+  XCTAssertNotNil(self.polylineController.polyline.spans);
```

Comment:
can you assert the actual values (rather than just asserting not nil)?

Reply:
![image](https://github.com/flutter/packages/assets/22373191/785a10b3-be29-48b3-b4a9-fe374aed4126)


I think that's not possible. Spans is of type `NSArray<GMSStyleSpan *>`
But as you can see in the screenshot, a single span itself doesn't have any observable props

Reply:
does `isEqual` work?

Reply:
I'm comparing just the first value of the array to illustrate the issue. I added a new `pattern` variable which should have the same value as that what's non nil



```
- (void)testSetPatterns {
  NSArray<GMSStrokeStyle *> *styles = @[
    [GMSStrokeStyle solidColor:[UIColor clearColor]], [GMSStrokeStyle solidColor:[UIColor redColor]]
  ];

  NSArray<NSNumber *> *lengths = @[ @10, @10 ];

  XCTAssertNil(self.polylineController.polyline.spans);

  [self.polylineController setPattern:styles lengths:lengths];

  NSArray<GMSStyleSpan *> *pattern =
      GMSStyleSpans(self.polylineController.polyline.path, styles, lengths, kGMSLengthRhumb);

//    XCTAssertNotNil(self.polylineController.polyline.spans);
  XCTAssertEqualObjects(self.polylineController.polyline.spans[0], pattern[0]);
}
```

On running this, I get an error that says
```
testSetPatterns: ((self.polylineController.polyline.spans[0]) equal to (pattern[0])) failed: ("<GMSStyleSpan: 0x600000a3ab80>") is not equal to ("<GMSStyleSpan: 0x600000b0fc60>")
```

<img width="381" alt="image" src="https://github.com/flutter/packages/assets/22373191/a6b79d2a-aa5c-4b13-8041-53dc7ec5c7d6">

<img width="397" alt="image" src="https://github.com/flutter/packages/assets/22373191/ed75716c-4618-4753-885e-649600fe31bf">
<br><br><br>
Looking at the segments in the debugger they do have the same value thought (Length of segment and color)
You can see that in the screenshots above and below. <br>
<img width="392" alt="image" src="https://github.com/flutter/packages/assets/22373191/7193cb01-1c46-4051-8a25-71e8fcd13b13">

<img width="446" alt="image" src="https://github.com/flutter/packages/assets/22373191/33e9caaa-64ac-45b1-9789-fb9c84ab4def">

Reply:
The default implementation of `isEqual:` is just pointer equality, so it's not surprising that it doesn't work here; GMS would have had to explicitly implement it for this object.

Reply:
So just stick to checking non nil for now?

Reply:
If it's not possible to check the actual value, nothing we can do. Can you add a comment there explaining this?

Reply:
Alright done!